server/controllers/index.js

```diff
@@ -38,22 +25,52 @@ exports.getNearbyBuildings = asyncHandler( async (req, res, next) => {
       }
     },
     workspaceId: { $ne: origin.workspaceId, $lte: 100, $gte: 1 }
-  });
+  })
+    .then(() => {
```

Comment:
Since the controller method is wrapped by the `async` keyword, we can eliminate the use of `.then().catch()` blocks.

```
router.get('/:workspaceId', async (req, res) => {
    try {
        const nearbyWorkspaces = await WorkspaceLocation.find({ query logic });
        if (nearbyWorkspaces.length === 0) return res.status(404).json({ msg: 'No nearby workspaces found' });
            // just an example, I am not exactly sure what the structure of the data returned is
        res.status(200).json({origin, nearbyWorkspaces});
    } catch (error) {
        handle error
    }
})
```

Reply:
This is an odd issue at the moment, and the quick then and catch block was an easy but not great solution. I'm actually using the catch on a partial success because two modals are updated in mongoose here in this one function. I'm gonna need to break both those out, so there aren't tied together. One modal is straightforward, but the other one is a specific index field that I haven't yet understood.